chain/network/src/peer/peer_actor.rs

```diff
@@ -651,22 +652,22 @@ impl PeerActor {
             ConnectingStatus::Outbound { handshake_spec: spec, .. } => {
                 if handshake.protocol_version != spec.protocol_version {
                     warn!(target: "network", "Protocol version mismatch. Disconnecting peer {}", handshake.sender_peer_id);
-                    ctx.stop();
+                    self.stop(ctx, None);
```

Comment:
semantically, it might reed a bit better with `stop(ctx, Ok(())) / stop(ctx, Err(ReasonForBan::Abuseive))`

Reply:
Done, in fact I've added a dedicated enum (ClosingReason), so that I can distinguish the reason of connection closing in tests.